action/main.go

```diff
@@ -81,6 +82,39 @@ func run(ctx context.Context) error {
 		slog.Bool("input/interactive", CLI.Build.Interactive),
 	)
 
+	// Check if submodules were initialized
+	// If they are uninitialized, just print warning to the user, ignore all errors
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	repo, err := git.PlainOpen(pwd)
+	if err == nil {
+		worktree, err := repo.Worktree()
+		if err != nil {
+			goto submodule_out
+		}
+		submodules, err := worktree.Submodules()
+		if err != nil {
+			goto submodule_out
+		}
+		status, err := submodules.Status()
+		if err != nil {
+			goto submodule_out
+		}
+		pattern := regexp.MustCompile(`(?m)^\-.*`)
+		matches := pattern.FindAllString(status.String(), -1)
+		for _, v := range matches {
+			patterSub := regexp.MustCompile(`^\-[\d\w]* `)
```

Comment:
Generally the PR looks fine but as I'm not sure how this regex check let's us determine that a submodule is initialized, could you briefly explain it either in a comment on this PR or in code?

Reply:
Oh, of course, my bad. According to the [documentation](https://git-scm.com/docs/git-submodule#Documentation/git-submodule.txt-status--cached--recursive--ltpathgt82308203), you can run `git submodule status` command to get a list of submodules.

For example:
```
 6f718d2ee5ea0f8ea2e993130e6cc6e90e9d0298 coreboot (remotes/origin/coreboot_tcc)
-f03a9663631535f298c35c4cc6bccd3a543abdf7 edk2 (edk2-stable201903-5320-gf03a966363)
```

If the line starts with dash (`-`), the submodule is not initialized. In the example above that would be `edk2`.

Reply:
Just added a link to documentation page into the code.